Library/Homebrew/cask/cmd/upgrade.rb

```diff
@@ -40,68 +40,81 @@ def run
         end
         puts cask_upgrades.join(", ")
 
+        caught_exceptions = []
         outdated_casks.each do |old_cask|
-          odebug "Started upgrade process for Cask #{old_cask}"
-          raise CaskUnavailableError.new(old_cask, "The Caskfile is missing!") if old_cask.installed_caskfile.nil?
+          begin
+            upgrade_cask(old_cask)
+          rescue CaskError => e
+            caught_exceptions << e
```

Comment:
Maybe collect the cask and the corresponding exception instead of just the exception.

Reply:
What should be done with the collected casks?

Reply:
I was just thinking about outputting the name and the corresponding exception.

Reply:
I think each exception is initialized with a reference to the cask that caused the exception, and `to_s` on the exception gives the cask name.